packages/string-store/src/lib/data/UnalignedBufferReader.ts

```diff
@@ -0,0 +1,75 @@
+import { Bit, Bit2, Bit4, IUnalignedBuffer } from './IUnalignedBuffer';
+
+export class UnalignedBufferReader extends IUnalignedBuffer {
+	public readBit(): Bit {
+		return this.read(1n) as Bit;
+	}
```

Comment:
Consider using a helper generator instead of re-implementing the same function innumerable times.
This could be done as follows, for instance:

```ts
function readBitX<T>(this: UnalignedBuffer, x: Size) {
    return () => {
        return this.read(x) as T;
    }
}
```
And then applying the generated functions to the class. This is as close to a macro as TypeScript will allow, but more readable and convenient.